File: core/server/project/commonBase/src/main/java/com/home/commonBase/scene/base/BulletLogicBase.java
package com.home.commonBase.scene.base;

import com.home.commonBase.config.game.BulletConfig;
import com.home.commonBase.config.game.BulletLevelConfig;
import com.home.commonBase.data.scene.base.BulletData;

public class BulletLogicBase extends SceneObjectLogicBase
{
	protected Bullet _bullet;
	
	protected Unit _unit;
	
	protected BulletData _data;
	
	protected BulletConfig _config;
	
	protected BulletLevelConfig _levelConfig;
	
	@Override
	public void setObject(SceneObject obj)
	{
		super.setObject(obj);
		
		_bullet=(Bullet)obj;
	}
	
	@Override
	public void init()
	{
		super.init();
		
		_unit=_bullet.getUnit();
		
		BulletData data=_data=_bullet.getData();
		
		_config=BulletConfig.get(data.id);
		_levelConfig=BulletLevelConfig.get(data.id,data.level);
	}
	
	@Override
	public void dispose()
	{
		super.dispose();
		
		_unit=null;
		_data=null;
		_config=null;
		_levelConfig=null;
	}
}

File: core/client/game/src/commonGame/scene/base/BulletLogicBase.cs
using System;
using ShineEngine;

/// <summary>
/// 
/// </summary>
public class BulletLogicBase:SceneObjectLogicBase
{
	protected Bullet _bullet;

	protected Unit _unit;

	protected BulletData _data;

	protected BulletConfig _config;

	protected BulletLevelConfig _levelConfig;

	public override void setObject(SceneObject obj)
	{
		base.setObject(obj);

		_bullet=(Bullet)obj;
	}

	public override void init()
	{
		base.init();

		_data=_bullet.getData();
		_unit=_bullet.getUnit();
		_config=BulletConfig.get(_data.id);
		_levelConfig=BulletLevelConfig.get(_data.id,_data.level);
	}

	public override void dispose()
	{
		base.dispose();

		_data=null;
		_unit=null;
		_config=null;
		_levelConfig=null;
	}
}